analytic_engine/src/instance/mod.rs

```diff
@@ -85,13 +85,13 @@ impl Spaces {
     }
 }
 
-pub struct SpaceStore<Wal, Meta> {
+pub struct SpaceStore {
     /// All spaces of the engine.
     spaces: RwLock<Spaces>,
     /// Manifest (or meta) stores meta data of the engine instance.
-    manifest: Meta,
+    manifest: ManifestImpl,
```

Comment:
This should be a trait obj like `ManifestRef`. But I don't know when we will add a second impl 🤨 So it is ok to keep `ManifestImpl` here (to me)

Reply:
Yes, it should be `ManifestRef`. And it is not hard to do that, we need remove associate type --- Error. Instead, `Box<dyn std::error::Error>` or the same Error for all concrete Manifest can be alternatives.
```rust
pub trait Manifest: fmt::Debug {
    type Error: std::error::Error + Send + Sync + 'static;
    ...
}
```

Reply:
Addressed.